main/pipeline/pipeline.cc

```diff
@@ -915,11 +917,19 @@ ast::ParsedFilesOrCancelled resolve(unique_ptr<core::GlobalState> &gs, vector<as
 }
 
 ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<ast::ParsedFile> what,
-                                      const options::Options &opts, WorkerPool &workers) {
+                                      const options::Options &opts, WorkerPool &workers, bool cancelable,
+                                      optional<shared_ptr<core::lsp::PreemptionTaskManager>> preemptionManager) {
```

Comment:
I could make this non-optional and default to a new preemption manager (so that preemption will never actually occur), but that would add the overhead of a reader lock acquisition to every file.